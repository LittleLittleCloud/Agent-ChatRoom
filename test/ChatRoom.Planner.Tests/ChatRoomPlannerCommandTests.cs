using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using Spectre.Console.Testing;
using Xunit;
using ChatRoom.Client.Tests;
using ChatRoom.SDK;
using Microsoft.Extensions.DependencyInjection;
using FluentAssertions;

namespace ChatRoom.Planner.Tests;

public class ChatRoomPlannerCommandTests : IClassFixture<EmptyChatRoomClientFixture>, IClassFixture<ChatRoomPlannerFixture>
{
    private readonly EmptyChatRoomClientFixture _fixture;
    private readonly ChatRoomPlannerFixture _plannerFixture;
    private readonly ChatPlatformClient _client;

    public ChatRoomPlannerCommandTests(EmptyChatRoomClientFixture fixture, ChatRoomPlannerFixture plannerFixture)
    {
        _fixture = fixture;
        _plannerFixture = plannerFixture;
        _client = plannerFixture.Command.ServiceProvider?.GetRequiredService<ChatPlatformClient>() ?? throw new InvalidOperationException("Failed to get ChatPlatformClient.");
    }

    [Fact]
    public async Task ItRegisterAndUnregisterAgentsToRoomAsync()
    {
        var agents = await _client.GetRoomMembers();
        var names = agents.Select(a => a.Name).ToList();
        names.Should().Contain("react-planner");

        var orchestrators = await _client.GetOrchestrators();
        orchestrators.Should().Contain("ReactPlanningOrchestrator");
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public async Task ItShowHelperTextTestAsync()
    {
        var app = new CommandAppTester();
        app.SetDefaultCommand<ChatRoomPlannerCommand>(ChatRoomPlannerCommand.Description);

        var result = await app.RunAsync("--help");

        Approvals.Verify(result.Output);
    }
}
