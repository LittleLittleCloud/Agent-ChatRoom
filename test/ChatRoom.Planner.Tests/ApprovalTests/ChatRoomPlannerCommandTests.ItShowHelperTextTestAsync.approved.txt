DESCRIPTION:
This package contains planners and corresponding orchestrators for chatroom.

The following planners are available:
- react-planner: a ReAct agent for multi-agent planning.

USAGE:
    testhost.dll [OPTIONS]

OPTIONS:
    -h, --help               Prints help information
    -v, --version            Prints version information
    -c, --config <CONFIG>    Configuration file